### PR TITLE
Support request-level asynchronous KV cache loading

### PR DESCRIPTION
**Pull request by lc1314555**
## Purpose
 Add request-level asynchronous KV cache loading for vLLM UCMDirectConnector.
 While one request is waiting for external KV data, vLLM can process other runnable requests and resume the waiting request after loading completes, reducing head-of-line blocking.
## Modifications 
 - Add the use_request_async configuration.
  - Record the UCM-to-vLLM block mapping after GPU block allocation.
  - Submit KV load tasks without blocking model execution.
  - Poll load completion through get_finished().
  - Prevent duplicate loads after rescheduling or preemption.
  - Report load failures and invalidate affected GPU blocks.
  - Add unit tests for dispatch, completion, duplicate prevention, and failure handling.
  - Currently support only UCMDirectConnector with a single KV cache group, PP=1, and without CP or layerwise loading.

### CODE REVIEW

File: ucm/integration/vllm/ucm_connector.py [hunk 1]
```diff
@@ -1174,6 +1182,28 @@ def __init__(
         self._invalid_block_ids: set[int] = set()
         self._async_dump_req_ids: set[str] = set()
         self._pending_dump_tasks: list[PendingDumpTask] = []
+        request_async_configured = bool(
+            self.launch_config.get("use_request_async", False)
```
Comment (qyh111) [inline ucm/integration/vllm/ucm_connector.py]: 建议修改为把参数值修改为use_request_async_load

**Review by qyh111 — APPROVED** [review]

**Review by SuperMarioYL — APPROVED** [review]
